README.md

```diff
@@ -207,6 +207,40 @@ common debug flow would involve:
 Another option is to change the scheme to "Wait for executable to be launched",
 then run the build product in Terminal.
 
+### Swift Toolchains
+
+#### Building
+
+Swift toolchains are created using the script
+[build-toolchain](https://github.com/apple/swift/blob/master/utils/build-toolchain). This
+script is used by swift.org's CI to produce snapshots and can allow for one to
+locally reproduce such builds for development or distribution purposes. E.x.:
+
+```
+  $ ./utils/build-toolchain $TOOLCHAIN_PREFIX
+```
+
+Beyond building the toolchain, ``build-toolchain`` also supports the following
+(non-exhaustive) set of useful options::
+
+- ``--dry-run``: Perform a dry run build. This is off by default.
+- ``--test``: Test the toolchain after it has been compiled. This is off by default.
+- ``--distcc``: Use distcc to speed up the build by distributing the c++ part of the swift build.
+
+More options may be added over time. Please pass ``--help`` to
+``build-toolchain`` to see the full set of options.
+
+#### Installing into Xcode
+
+On macOS if one wants to install such a toolchain into Xcode:
+
+1. Untar and copy the toolchain to `/Library/Developer/Toolchains/`.
```

Comment:
Users can untar the toolchain in to two location. 

1. `~/Library/Developer/Toolchains/`
2. `/Library/Developer/Toolchains/`

We should also povide command to untar. 

for example: 

`tar -zxf swift-osx.tar.gz --directory ~/`